src/json.hpp

```diff
@@ -8011,6 +8013,7 @@ class basic_json
         json_value(number_float_t v) noexcept : number_float(v) {}
         /// constructor for empty values of a given type
         json_value(value_t t)
+            : object(nullptr)
```

Comment:
If this is required, then maybe we can remove the lines

```cpp
object = nullptr;  // silence warning, see #821
```

below which have been added when discussing #821.

Reply:
Related: line 8073 is now unreachable due to the case added at line 8062.

Reply:
That case was always unreachable and was only added to get the hash `961c151d2e87f2686a955a9be24d316f1362bf21` into binaries using the library.

Reply:
Interesting.  Perhaps a comment is in order the.

Reply:
I think this can be removed -- it was bad icc check.

Reply:
Thanks. Could you remove the mentioned line above please?